includes/admin/class-wc-admin-addons.php

```diff
@@ -40,36 +40,97 @@ public static function get_featured() {
 	}
 
 	/**
-	 * Get sections for the addons screen
+	 * Build transient name
 	 *
-	 * @return array of objects
+	 * @param  string $category
+	 * @param  string $term
+	 * @param  string $country
+	 *
+	 * @return string transient name
 	 */
-	public static function get_sections() {
-		if ( false === ( $sections = get_transient( 'wc_addons_sections' ) ) ) {
-			$raw_sections = wp_safe_remote_get( 'https://d3t0oesq8995hv.cloudfront.net/addon-sections.json', array( 'user-agent' => 'WooCommerce Addons Page' ) );
-			if ( ! is_wp_error( $raw_sections ) ) {
-				$sections = json_decode( wp_remote_retrieve_body( $raw_sections ) );
+	public static function build_transient_name( $category, $term, $country ) {
+		$transient_name = 'wc_addons_extensions';
+		if ( $category ) {
+			$transient_name = $transient_name . '_category_' . $category;
+		}
+		if ( $term ) {
+			$transient_name = $transient_name . '_term_' . $term;
+		}
+		if ( $country ) {
+			$transient_name = $transient_name . '_country_' . $country;
+		}
+		return $transient_name;
```

Comment:
I think we should hash these for two reasons.

First – there's a 191 character limit on `option_name` in the `wp_options` table, so searching for really long strings (or copy/paste errors, or search by description, mostly edge cases I guess...) will truncate names silently on most MySQL/Maria servers.

Second – with `$term` we're piping user input directly into `set_transient()`, probably not a huge deal since options in WordPress can actually be named `'; DROP TABLE wp_users; --` and still work fine, but hey. Better safe than SQLi.

What I would do is probably something along the lines of:

```
public static function build_transient_name( $args ) {
    return 'wc_addons_extensions_' . md5( json_encode( $args ) ) );
}
```

Then you can call it with `array( 'term' => 'foo', 'category' => 'shipping-methods' )` or whatever.